xls/build_rules/xls_utilities.bzl

```diff
@@ -306,3 +307,80 @@ Examples:
     implementation = _proto_data_impl,
     attrs = _proto_data_attrs,
 )
+
+def _proto_to_dslx_impl(ctx):
+    """The implementation of the 'proto_to_dslx' rule.
+
+    Args:
+      ctx: The current rule's context object.
+
+    Returns:
+      DefaultInfo provider
+    """
+    schema_sources = ctx.attr.proto_schema[ProtoInfo].direct_sources
+    schema = schema_sources[0]
+
+    proto2dslx = ctx.executable._proto_to_dslx_tool
+    dslx_filename = get_output_filename_value(
+        ctx,
+        "dslx_file",
+        ctx.attr.name + ".x",
```

Comment:
should we allow to overwrite the output path?

Reply:
Isn't the `dslx_file` attribute enough? It's used when it's specified in the rule, otherwise it defaults to `ctx.attr.name + ".x"`

Reply:
Resolution SGTM we can expand as we find a need.